### PR TITLE
Automatically show user tutorial 

### LINKED ISSUE #727: Automatically show tutorial when user first signs up

### LINKED ISSUE #763: deleted my account and remade a new one with the same username - my cover photo from the last account is still there

### LINKED ISSUE #769: Creating privately to pack should navigate to PackOpen
Getting an error here - @Nash0x7E2 can you take a look?

### LINKED ISSUE #771: Optimize PhotoExpression loading
On the latest dev, I notice we display the lower res image preview but show a grey placeholder while the full image loads from the network. 

I think we can optimize this a little more and display the lower res image until the full image loads. At that point, we can fade/switch the displayed image to the higher resolution image.

### PR DESCRIPTION
**Pull request by Nash0x7E2**
This PR touches on a few issues, the primary focus was on automatically showing the user tutorial when the user first signs up. In addition to this, the following issues also touched:

- Closes #727
- Closes #763
- Closes #769 (Bloc context not available)
- Closes #771

### CODE REVIEW

**Comment by Nash0x7E2** [comment]
@orestesgaolin Following our conversation during today's dev call, I created a new repo specifically for controlling whether we should show the tutorial or not. Lmk if there's any other optimization you think we can make in this area.

**Comment by Nash0x7E2** [comment]
@sunyatax I experimented with showing the lower res image in `PhotoOpen` while we wait on the network to load the full image. Since the small thumbnail is already cached, it results in a smoother UX since users are not presented with another loader. Let me know if you have any suggestions of further improving this :)

**Comment by sunyatax** [comment]
Great job, Nash. Nice touch on the PhotoOpen placeholder.

One thing... noticed that the tutorial still shows up after log in. Can you enable the tutorial only after the user signs up for the 1st time and disable for each subsequent log in? Maybe we have to get some sort of endpoint in the API involved here, instead of relying on local state.

**Comment by orestesgaolin** [comment]
Yeah, I think that without support from api we won't be able to hide the tutorial on subsequent logins. We clean the user data on logout. Maybe Josh can expose some basic endpoint for that.

File: lib/screens/packs/packs_list_appbar.dart [hunk 1]
```diff
@@ -17,6 +19,21 @@ class PacksListAppBar extends StatelessWidget {
...
+class _PacksListAppBarState extends State<PacksListAppBar> {
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final repo = Provider.of<OnBoardingRepo>(context);
+    if (repo.showPackTutorial) {
+      showTutorial();
+      repo.setViewed(HiveKeys.kShowPackTutorial, false);
+    }
+  }
```
Comment (orestesgaolin) [inline lib/screens/packs/packs_list_appbar.dart:35]: Just an idea, but maybe...

```
OnBoardingRepo.of(context).maybeShow(Tutorial.showPack, showTutorial);
```

```
void maybeShow(Tutorial type, VoidCallback onShow){
if (showPackTutorial){
onShow();
setViewed(...)
}
}
```

Comment (orestesgaolin) [reply lib/screens/packs/packs_list_appbar.dart:35]: And then replace it everywhere where it's used

Comment (Nash0x7E2) [reply lib/screens/packs/packs_list_appbar.dart:35]: Good idea, I’ll update

**Pull request merged**